### PR TITLE
CON-2884 Markdown Config Protector Exercise

### PR DESCRIPTION
**Pull request by zanninso**
CON-2884

### CODE REVIEW

File: subjects/java/checkpoints/config-protector/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+## Config Protector
+
+### Instructions
+
+Create a class `ConfigProtector` that provides a method to hide sensitive data in a configuration file using regex. The method should replace sensitive values with asterisks. The configuration file will be provided as a string, and the keys for the sensitive data will be given in a list.
```
Comment (MSilva95) [inline subjects/java/checkpoints/config-protector/README.md:5]: Maybe we could highlight the word Regex and the types of the files as well, the string and the list.

File: subjects/java/checkpoints/config-protector/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+```
+
+### Usage
+
+Here is a possible `ExerciseRunner.java` to test your class:
+
+```java
+import java.util.Arrays;
+
```
Comment (MSilva95) [inline subjects/java/checkpoints/config-protector/README.md:25]: It's missing this import:
`import java.util.List;`

File: subjects/java/checkpoints/config-protector/ExerciseRunner.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import java.util.Arrays;
+
```
Comment (MSilva95) [inline subjects/java/checkpoints/config-protector/ExerciseRunner.java:2]: It's missing this import:
`import java.util.List;`

File: subjects/java/checkpoints/config-protector/ExerciseRunner.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        String configFile2 = "apiKey=12345\napiSecret=abcdef\nendpoint=https://api.example.com\n";
+        List<String> sensitiveKeys2 = Arrays.asList("apiKey", "apiSecret");
+        System.out.println("Protected Config 2:\n" + protector.hideSensitiveData(configFile2, sensitiveKeys2));
+
+        // Test case 3
+        String configFile3 = "username=user\npassword=pass\n";
+        List<String> sensitiveKeys3 = Arrays.asList("username", "password");
+        System.out.println("Protected Config 3:\n" + protector.hideSensitiveData(configFile3, sensitiveKeys3));
+    }
+}
```
Comment (MSilva95) [inline subjects/java/checkpoints/config-protector/ExerciseRunner.java:22]: Needs to be formatted, last line here.

**Review by MSilva95 — CHANGES_REQUESTED** [review]

File: subjects/java/checkpoints/config-protector/README.md (hunk 1 above)
Comment (zanninso) [reply subjects/java/checkpoints/config-protector/README.md:5]: what do you mean by saying the types of the files?

File: subjects/java/checkpoints/config-protector/README.md (hunk 2 above)
Comment (zanninso) [reply subjects/java/checkpoints/config-protector/README.md:25]: fixed

File: subjects/java/checkpoints/config-protector/ExerciseRunner.java (hunk 3 above)
Comment (zanninso) [reply subjects/java/checkpoints/config-protector/ExerciseRunner.java:2]: fixed

File: subjects/java/checkpoints/config-protector/ExerciseRunner.java (hunk 4 above)
Comment (zanninso) [reply subjects/java/checkpoints/config-protector/ExerciseRunner.java:22]: fixed

File: subjects/java/checkpoints/config-protector/README.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+## Config Protector
+
+### Instructions
+
+Create a class `ConfigProtector` that provides a method to hide sensitive data in a configuration file using `Regex`. The method should replace sensitive values with asterisks. The configuration file will be provided as a string, and the keys for the sensitive data will be given in a list.
+
+### Expected Class
+
+```java
+import java.util.List;
```
Comment (nprimo) [inline subjects/java/checkpoints/config-protector/README.md:10]: Shouldn't these imports be included as well?
```java
import java.util.regex.Matcher;
import java.util.regex.Pattern;
```

**Review by nprimo — CHANGES_REQUESTED** [review]
I left some comments inline. Additionally, it might be that more info needs to be added to the subject due to a comment I left in the [tests PR](https://github.com/01-edu/java-tests/pull/58)

File: subjects/java/checkpoints/config-protector/README.md (hunk 5 above)
Comment (zanninso) [reply subjects/java/checkpoints/config-protector/README.md:18]: yeah, done.

**Review by nprimo — CHANGES_REQUESTED** [review]

File: subjects/java/checkpoints/config-protector/README.md (hunk 1 above)
Comment (MSilva95) [reply subjects/java/checkpoints/config-protector/README.md]: "... provided as a `string`, and the keys for the sensitive data will be given in a `list`."

Comment (zanninso) [reply subjects/java/checkpoints/config-protector/README.md]: done

File: subjects/java/checkpoints/config-protector/README.md [hunk 6]
```diff
@@ -29,7 +39,7 @@ public class ExerciseRunner {
         ConfigProtector protector = new ConfigProtector();
 
         // Test case 1
-        String configFile1 = "username=admin\npassword=secret\nhost=localhost\n";
+        String configFile1 = "username=admin\n=localhost\n";
```
[the remark this answers is not in the document]
Comment (zanninso) [reply subjects/java/checkpoints/config-protector/README.md]: done

**Review by nprimo — APPROVED** [review]
LGTM

**Review by MSilva95 — APPROVED** [review]
LGTM!

**Pull request merged**